WSLGd/FontMonitor.cpp

```diff
@@ -43,39 +43,71 @@ wslgd::FontFolder::~FontFolder()
     }
 }
 
-void wslgd::FontFolder::ExecuteShellCommand(const char *cmd)
+bool wslgd::FontFolder::ExecuteShellCommand(std::vector<const char*>&& argv)
 {
     bool success = false;
+    int childPid = -1, waitPid = -1;
+    std::string cmd;
+
+    try {
+        THROW_LAST_ERROR_IF((childPid = fork()) < 0);
+        if (childPid == 0) {
+            /* move this process to own process group to avoid interfere with Process Monitor */
+            THROW_LAST_ERROR_IF(setpgid(0, 0) < 0);
+            THROW_LAST_ERROR_IF(execvp(argv[0], const_cast<char *const *>(argv.data())) < 0);
+        } else if (childPid > 0) {
+            /* move child to own process group to avoid interfere with Process Monitor */
+            THROW_LAST_ERROR_IF(setpgid(childPid, childPid) < 0);
+        }
+    }
+    CATCH_LOG();
+
+    // Ensure that the child process exits.
+    if (childPid == 0) {
+        _exit(1);
+    }
+
+    if (childPid > 0) try {

```

Comment:
this is always true.

Reply:
right, I can remove the if statement but leaving only try.

Reply:
Actually, I think it is not, if fork is failed, it caught by catch at line 63, and it continues, thus it can come here with -1.

Reply:
Hmm - if fork fails we should probably just take down the process (move it out of the try).

Reply:
But this failing to execute xset isn't so fatal though, and if WSLGd existed, there is no auto relaunch of it.